### ISSUE TITLE
Inaccurate Ruby logger code snippets and line numbers

### ISSUE THREAD

**Issue by elsapet** [Bearer/bearer-rules#54]
## Description & Reproduction

The console report for Ruby lang and Rails logger findings is confusing. 
The line number given corresponds to the sensitive data and not the inclusion of this sensitive data within the logger call. The code snippet (and so, presumably, the parent line numbers) correspond to the logger call itself, but since the line number is not within the parent line number, the snippet shows without any content or highlight.

### Repro steps

1. Run a scan on the following code snippet (in a file): 

```ruby
class User < ApplicationRecord
  def get_user(email)
    user_params = {
      primary_email: UserEmail.new(email: email, primary: true)
    }

    Rails.logger.warn(
      "New User (user_id: #{user.id}) Params: #{user_params}"
    )

    logger.error(
      "New User (user_id: #{user.id}) Params: #{user_params}"
    )

    user
  end
end
```

2. Examine the report given

## Expected Behavior

Code snippet should include relevant pieces of code

## Actual Behavior

Line number is not accurate
Code snippet is empty

![Screenshot 2023-05-15 at 13 17 33](https://github.com/Bearer/bearer-rules/assets/4560746/598448ee-794f-44fb-b129-942817607867)

## Your Environment

- Operating System and version: 
- Output of 'bearer version':

```
bearer version: 1.6.1
sha: 743e8d8a246db1225670f0e34766cfc139cb3d5a
```

Note also seen on 
```
bearer version: main
sha: bbf6cc415923f13742da6d0399f26d92190ec47f```

**Comment by elsapet** [comment]
@cfabianski @didroe is this desired behaviour with the line numbers? If yes, we probably need to handle it a bit better in the code snippet 🤔

**Comment by cfabianski** [comment]
I think updating the pattern from `Rails.logger.$<METHOD>($<DATA_TYPE>)` to `$<!>Rails.logger.$<METHOD>($<DATA_TYPE>)` should do the trick. 
What do you think?

**Comment by elsapet** [comment]
Ah thanks @cfabianski - alas no change with `$<!>Rails.logger.$<METHOD>($<DATA_TYPE>)`, but `Rails.logger.$<METHOD>($<!>$<DATA_TYPE>)` gave the following output; I'm not sure if it is better than what we currently have 🤔 

<img width="809" alt="Screenshot 2023-05-19 at 16 56 21" src="https://github.com/Bearer/bearer-rules/assets/4560746/d5b59432-71a6-48a2-a1d9-affe355d6a1d">

Maybe I am misunderstanding how `$<!>` functions, but for me, the main issue is how to render the code snippets if the line number is not within the parent line number.
